src/aiida_workgraph/utils/__init__.py

```diff
@@ -332,32 +329,12 @@ def get_processes_latest(
                     "mtime": link.node.mtime,
                 }
     elif item_type == "task":
-        node_names = [node_name] if node_name else []
-        if node_name:
-            projections = [
-                f"extras._task_state_{node_name}",
-                f"extras._task_process_{node_name}",
-            ]
-        else:
-            projections = []
-            process = aiida.orm.load_node(pk)
-            node_names = [
-                key[12:]
-                for key in process.base.extras.keys()
-                if key.startswith("_task_state")
-            ]
-            projections = [f"extras._task_state_{name}" for name in node_names]
-            projections.extend([f"extras._task_process_{name}" for name in node_names])
-        qb = QueryBuilder()
-        qb.append(WorkGraphEngine, filters={"id": pk}, project=projections)
-        # print("projections: ", projections)
-        results = qb.all()
-        # change results to dict
-        results = dict(zip(projections, results[0]))
-        # print("results: ", results)
-        for name in node_names:
-            state = results[f"extras._task_state_{name}"]
-            task_process = deserialize_safe(results[f"extras._task_process_{name}"])
+        task_states = node.task_states
+        task_processes = node.task_processes
+        task_names = [task_name] if task_name else task_states.keys()
+        for name in task_names:
+            state = task_states[name]
+            task_process = deserialize_safe(task_processes.get(name, ""))
```

Comment:
Note for me: Because we have now store the information in the node we can retrieve this information directly from the node instead of querying the database